utils/pagination.py

```diff
@@ -0,0 +1,63 @@
+from typing import List, Tuple
+import discord
+from discord.ext import commands
+from discord import ui
+
+class Pagination(ui.View):
+    def __init__(self, ctx: commands.Context, embeds: List[discord.Embed]):
+        super().__init__(timeout = 60)
+        self.ctx = ctx
+        self.embeds = embeds
+        self.current_page = 0
+
+    @ui.button(emoji=u"\u23EA", style=discord.ButtonStyle.blurple)
+    async def first_page(self, interaction: discord.Interaction, button):
+        self.current_page = 0
+        await self.update_page(interaction)
+    
+    @ui.button(emoji=u"\u2B05", style=discord.ButtonStyle.blurple)
+    async def before_page(self, interaction: discord.Interaction, button):
+        if self.current_page > 0:
+            self.current_page -= 1
+            await self.update_page(interaction)
+    
+    @ui.button(emoji=u"\u27A1", style=discord.ButtonStyle.blurple)
+    async def next_page(self, interaction: discord.Interaction, button):
+        if self.current_page < len(self.embeds) - 1:
+            self.current_page += 1
+            await self.update_page(interaction)
+    
+    @ui.button(emoji=u"\u23E9", style=discord.ButtonStyle.blurple)
+    async def last_page(self, interaction: discord.Interaction, button):
+        self.current_page = len(self.embeds) - 1
+        await self.update_page(interaction)
+    
+    async def update_page(self, interaction: discord.Interaction):
+        if self.current_page == 0:
```

Comment:
It might be more concise to have
```py
for i in self.children: i.disabled = False
if self.current_page == 0:
    self.children[0].disabled = True
    self.children[1].disabled = True
if self.current_page == len(self.embeds) - 1:
    self.children[-1].disabled = True
    self.children[-2].disabled = True
```
I would isolate this logic into something like `async def update_buttons()`, and then call it as well as `await interaction.response.edit_message()` within each button's method. `update_buttons` could be called inside `return_paginated_embed_view` too, since it would disable all buttons when `self.current_page == 0 == len(self.embeds) - 1`.